### PR TITLE
[IMP] im_livechat: introduce PublicLivechatWindow/inputPlaceholder

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2928859

### CODE REVIEW

File: addons/im_livechat/static/src/models/public_livechat_window.js [hunk 1]
```diff
@@ -27,6 +26,18 @@ registerModel({
...
+    recordMethods: {
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeInputPlaceholder() {
+            if (this.messaging.livechatButtonView.inputPlaceholder) {
+                return this.messaging.livechatButtonView.inputPlaceholder;
+            }
+            return this.env._t("Say something");
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_window.js:38]: where does this come from?

**Review by oomsveta — APPROVED** [review]

File: addons/im_livechat/static/src/models/public_livechat_window.js (hunk 1 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_window.js:38]: I forgot to remove where it was used. You should see the negative diff that put this "Say something" as default.

File: addons/im_livechat/static/src/models/public_livechat_window.js [hunk 2]
```diff
@@ -36,5 +47,8 @@ registerModel({
             readonly: true,
             required: true,
         }),
+        inputPlaceholder: attr({
+            compute: '_computeInputPlaceholder',
+        }),
```
Comment (oomsveta) [inline addons/im_livechat/static/src/models/public_livechat_window.js:52]: alphabetical

Comment (alexkuhn) [reply addons/im_livechat/static/src/models/public_livechat_window.js]: Done

File: addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js [hunk 3]
```diff
@@ -39,9 +39,7 @@ const PublicLivechatWindow = Widget.extend({
     init(parent, messaging, thread, options) {
         this._super(parent);
         this.messaging = messaging;
-        this.options = _.defaults(options || {}, {
-            placeholder: _t("Say something"),
-        });
+        this.options = _.defaults(options || {});
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:42]: call to `_defaults` is now useless

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:42]: Not really, `options` could be `undefined`, so this `_.defaults()` makes in an empty object.

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js:42]: But actually never the case, so will just remove it

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/widgets/public_livechat_window/public_livechat_window.js]: Changes done

**Review by oomsveta — APPROVED** [review]

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/96690

**Pull request closed without merging**